pkg/collector/corechecks/servicediscovery/language/language.go

```diff
@@ -89,16 +89,16 @@ func (pi ProcessInfo) FileReader() (io.ReadCloser, bool) {
 	if strings.HasPrefix(fileName, "/") {
 		return findFile(fileName)
 	}
-	for _, env := range pi.Envs {
-		if key, val, _ := strings.Cut(env, "="); key == "PATH" {
-			paths := strings.Split(val, ":")
-			for _, path := range paths {
-				if r, found := findFile(path + string(os.PathSeparator) + fileName); found {
-					return r, true
-				}
+	if val, ok := pi.Envs["PATH"]; ok {
+		paths := strings.Split(val, ":")
+		for _, path := range paths {
+			if r, found := findFile(path + string(os.PathSeparator) + fileName); found {
```

Comment:
nit: (can be moved to a different PR as it is not relevant to the change you're introducing)
```suggestion
			if r, found := findFile(filepath.Join(path, fileName)); found {
```

Reply:
Leaving this change out of this PR since it's not strictly related to its purpose.